compiler-lib/src/lowering/amd64/live_variable_analysis.rs

```diff
@@ -0,0 +1,340 @@
+        for lir::CopyPropagation { src, dst } in &code.copy_in {
+            instrs.push(Instruction::Mov {
+                src: *src,
+                dst: dst.multislot(),
+            });
+        }
+        for instr in &code.body {
+            match instr {
+                lir::Instruction::Comment(_) => (),
+                _ => instrs.push(self.gen_instr(instr)),
+            }
+        }
+        for lir::CopyPropagation { src, dst } in &code.copy_out {
+            instrs.push(Instruction::Mov {
+                src: *src,
+                dst: dst.multislot(),
+            })
+        }
+        if let Some(leave) = code.leave.get(0) {
+            instrs.push(Instruction::Leave(leave.clone()));
+        }
+
+        instrs
+    }
+
+    fn gen_instr(&self, instr: &lir::Instruction) -> Instruction {
+        if let lir::Instruction::Call { .. } = instr {
+            Instruction::Call(FunctionCall::new(self.cconv, instr.clone()))
+        } else {
+            Instruction::Lir(instr.clone())
+        }
+    }
+
+    fn gen_queue(
+        &mut self,
+        end_block: Ptr<BasicBlock>,
+        visited: &mut HashSet<libfirm_rs::nodes::Block>,
+    ) {
+        self.queue.push_back(end_block);
+
+        for pred in end_block.pred_blocks() {
+            if visited.insert(pred.firm) {
+                self.gen_queue(pred, visited);
+            }
+        }
+    }
+}
+
+impl Instruction {
+    pub(super) fn src_operands(&self) -> Vec<lir::Operand> {
+        use super::{
+            function::FnInstruction,
+            lir::{Instruction::*, Leave::*},
+        };
+        match self {
+            Instruction::Lir(lir) => match lir {
+                Binop { src1, src2, .. } | Div { src1, src2, .. } | Mod { src1, src2, .. } => {
+                    vec![*src1, *src2]
+                }
+                Unop { src, .. } | Conv { src, .. } => vec![*src],
+                Call { .. } => vec![], // already converted
+                StoreMem {
+                    src,
+                    dst: lir::AddressComputation { base, index, .. },
+                } => {
+                    let mut ops = vec![*src, *base];
+                    match index {
+                        lir::IndexComputation::Zero => (),
+                        lir::IndexComputation::Displacement(op, _) => ops.push(*op),
+                    }
+                    ops
+                }
+                LoadMem {
+                    src: lir::AddressComputation { base, index, .. },
+                    ..
+                } => {
+                    let mut ops = vec![*base];
+                    match index {
+                        lir::IndexComputation::Zero => (),
+                        lir::IndexComputation::Displacement(op, _) => ops.push(*op),
+                    }
+                    ops
+                }
+                Comment(_) => vec![],
+            },
+            Instruction::Mov { src, .. } => vec![lir::Operand::try_from(*src).unwrap()],
```

Comment:
Isn't this also possible?

```suggestion
            Instruction::Mov { src, .. } => vec![src.try_into().unwrap()],
```

Reply:
Replaced it with a `From` impl and now use `(*src).into()`